Add text to figure

Tukop opened the pull request:
Modify textsize with strokesize
Modify textcolour with stroke color
Modify text with text attribute when textbox is the selected shape

Commit:
added snapgrid element to the project

Commit:
binded toolbar grid spinner to the snapgrid

Commit:
added grid view functionnalit

Commit:
now when updating the spacing spinner, the grid is updated

Commit:
now create shape snaps the new shape to the grid. **grid points and grid lines are not sync correctly**

Commit:
fix snapping mode on lastclickposition not being snap to the grid

Tukop committed:
Linked TextBox Object to new text attribute

Tukop committed:
Whoopsie

boscoa committed:
Added composite pattern to select multiple shapes.

Commit:
fix snapgrid canvas out-of-bounds

Commit:
remove debugger for grid

pspino commented on src/main/java/apaintus/controllers/CanvasController.java:
```diff
@@ -200,4 +200,20 @@ public void changed(ObservableValue<? extends Double> observableValue, Double ol
+
+        @Override
+        public void changed(ObservableValue<? extends String> observableValue, String oldValue, String newValue) {
+            if(activeShape != null && activeShape.isSelected() && drawnShapes.contains(activeShape)) {
```
does this means that the text can only change when a shape is selected from inside the attribute bar ?

Tukop replied:
I have not modified that part of the code, the only changes on that section is the spacing on the left

Commit:
made some changes to reduce the McCabe Cyclomatic Complexity

pgarneau commented on src/main/java/apaintus/controllers/AttributeController.java:
```diff
@@ -27,6 +24,8 @@
     @FXML
     private Spinner<Double> shapeHeight;
+    @FXML
+    private TextArea shapeText;
```
Est-ce que c'est vraiment un shapeText ou plus juste du text. On met pas du text sur des shapes

Tukop replied:
changed to text

pgarneau commented on src/main/java/apaintus/models/shapes/DrawableShape.java:
```diff
@@ -8,17 +8,20 @@
 public abstract class DrawableShape extends Shape {
     protected boolean selected = true;
     protected BoundingBox boundingBox;
+    protected String text;
```
This doesn't belong here. On en a parle

Tukop replied:
removed

Tukop committed:
Removed old code suggesting text in shapes

boscoa committed:
Added ungroup action. Moved vertices for loop inside selection box.

Ungroup button added to attributes panel. This will trigger an ActionEvent that clears the selectionBox and remove it from drawnShapes.

Commit:
added review suggested changes

Commit:
changes from review

Commit:
rename changed snapgrid package name as it was wrongly written

Commit:
white spaces

boscoa commented on src/main/java/apaintus/models/shapes/TextBox.java:
```diff
@@ -28,7 +27,8 @@ public TextBox(ShapeAttributes shapeAttributes) {
         super.update(shapeAttributes);
 
-        //text = shapeAttributes.getText();
+        if (shapeAttributes.getText() != null )
```
Est-ce que le textArea peut renvoyer null si rien n'est écrit?

Tukop replied:
non il envoit un string vide

boscoa replied:
Le if not null pourrait être retiré si tu mets un text avec withText dans la factory dans le cas d'une textShape. À ta discrétion.

boscoa commented on src/main/java/apaintus/models/shapes/TextBox.java:
```diff
@@ -17,8 +17,7 @@ private TextBox() {
 
-        text = new String("Enter text down below \n"
-        		+ "dab on the haters in the attributes");
+        text = new String("Enter text down below");
```
Ne serait-il pas mieux de setter la string dans la factory et ensuite de la get des shapeAttributes?

Commit:
moved activateSnapGrid to upper lines to avoid confusion of it to be living inside private classes of CanvasController

Commit:
removed useless constructor

Commit:
shit shat shot

pgarneau committed:
Merge pull request #9 from pgarneau/snapgrid

Snapgrid

boscoa committed:
Added translation and rotation to composite shapes

Commit:
fix a line that had x3 the same args in it instead of the right ones

boscoa committed:
Rebase onto master

boscoa committed:
Updated SelectionBox

Calls boundingBox update instead of update, preventing a new update on fields.
Removed update method, it was the same as DrawableShape

boscoa committed:
Merge pull request #10 from pgarneau/multiple_selection

Multiple selection

Tukop committed:
Save load application preferences (#8)

* added snapgrid element to the project

* binded toolbar grid spinner to the snapgrid

*  added grid view functionnalit

* now when updating the spacing spinner, the grid is updated

* now create shape snaps the new shape to the grid. **grid points and grid lines are not sync correctly**

* fix snapping mode on lastclickposition not being snap to the grid

* Added application preference save/load

The data saved isn't correct yet

* Added the strokeSize, strokeColor and fillColor to the preferences

* Added the lastLoad and lastSave repository preferences

Also added protection against empty text file when first launching

* fix snapgrid canvas out-of-bounds

* remove debugger for grid

* made some changes to reduce the McCabe Cyclomatic Complexity

* Added config.properties

* added review suggested changes

* changes from review

* rename changed snapgrid package name as it was wrongly written

* white spaces

* moved activateSnapGrid to upper lines to avoid confusion of it to be living inside private classes of CanvasController

* removed useless constructor

* shit shat shot

* Created a Preferences Service

Modified the code to abide by the new structure

* removed unused imports

* Added application preference save/load

The data saved isn't correct yet

* Added the strokeSize, strokeColor and fillColor to the preferences

* Added the lastLoad and lastSave repository preferences

Also added protection against empty text file when first launching

* Added config.properties

* Created a Preferences Service

Modified the code to abide by the new structure

* removed unused imports

*  fix a line that had x3 the same args in it instead of the right ones

* Added composite pattern to select multiple shapes.

* Added ungroup action. Moved vertices for loop inside selection box.

Ungroup button added to attributes panel. This will trigger an ActionEvent that clears the selectionBox and remove it from drawnShapes.

* Added translation and rotation to composite shapes

* Rebase onto master

* Updated SelectionBox

Calls boundingBox update instead of update, preventing a new update on fields.
Removed update method, it was the same as DrawableShape

* Changed the Application Preference Architecture

Removed lots of spaces

* Added application preference save/load

The data saved isn't correct yet

* Added the strokeSize, strokeColor and fillColor to the preferences

* Added the lastLoad and lastSave repository preferences

Also added protection against empty text file when first launching

* Added config.properties

* Created a Preferences Service

Modified the code to abide by the new structure

* removed unused imports

* Added application preference save/load

The data saved isn't correct yet

* Added the strokeSize, strokeColor and fillColor to the preferences

* Added the lastLoad and lastSave repository preferences

Also added protection against empty text file when first launching

* Added config.properties

* Created a Preferences Service

Modified the code to abide by the new structure

* removed unused imports

* Changed the Application Preference Architecture

Removed lots of spaces

* Prevent crash on first launch

pgarneau committed:
rebase
